rustler_tests/native/rustler_test/src/test_codegen.rs

```diff
@@ -85,6 +162,21 @@ pub fn untagged_enum_echo(untagged_enum: UntaggedEnum) -> UntaggedEnum {
     untagged_enum
 }
 
+#[derive(NifUntaggedEnum)]
+pub enum GenericUntaggedEnum<T, U> {
+    Foo(u32),
+    Bar(T),
+    Baz(U),
+    Bool(bool),
+}
+
+#[rustler::nif]
+pub fn generic_untagged_enum_echo(
+    generic_untagged_enum: GenericUntaggedEnum<GenericMap<i32, i32>, &str>,
+) -> GenericUntaggedEnum<GenericMap<i32, i32>, &str> {
+    generic_untagged_enum
+}
+
```

Comment:
A test with explicit lifetimes would be nice. Something along these lines maybe? Testing with lifetimes could be tricky, but a `'static` string might suffice.

```rust
#[derive(NifStruct)]
struct Slice<'a> {
  my_str: &'a str
}

#[rustler::nif]
pub fn from_a_string_slice() -> Slice {
  Slice { my_str: "hello world" }
}
```

Reply:
Do you mean an explicit lifetime together with a type parameter? I (hope I) haven't changed anything about the behaviour with lifetimes here. I'll send a short PR ahead that adds a lifetime test (since it seems there isn't one yet in `test_codegen.rs`) and rebase onto that, then we know that I didn't break them.

Reply:
Ok, seems like lifetimes are broken anyways, #428. This looks like an orthogonal issue to me.